src/sentry/incidents/endpoints/serializers/workflow_engine_incident.py

```diff
@@ -0,0 +1,116 @@
+from collections.abc import Mapping
+from typing import Any, ClassVar
+
+from django.contrib.auth.models import AnonymousUser
+
+from sentry.api.serializers import Serializer
+from sentry.api.serializers.models.incidentactivity import IncidentActivitySerializerResponse
+from sentry.incidents.endpoints.serializers.alert_rule import AlertRuleSerializerResponse
+from sentry.incidents.endpoints.serializers.incident import (
+    DetailedIncidentSerializerResponse,
+    IncidentSerializerResponse,
+)
+from sentry.incidents.models.incident import IncidentStatus, IncidentStatusMethod, IncidentType
+from sentry.models.groupopenperiod import GroupOpenPeriod
+from sentry.snuba.entity_subscription import apply_dataset_query_conditions
+from sentry.snuba.models import SnubaQuery
+from sentry.types.group import PriorityLevel
+from sentry.users.models.user import User
+from sentry.users.services.user.model import RpcUser
+
+
+class WorkflowEngineIncidentSerializer(Serializer):
+    priority_to_incident_status: ClassVar[dict[PriorityLevel, IncidentStatus]] = {
+        PriorityLevel.HIGH: IncidentStatus.CRITICAL,
+        PriorityLevel.MEDIUM: IncidentStatus.WARNING,
+        PriorityLevel.LOW: IncidentStatus.OPEN,
+    }
+
+    def __init__(self, expand=None):
+        self.expand = expand or []
+
+    def get_attrs(self, item_list, user, **kwargs):
+
+        results = {}
+        for open_period in item_list:
+            results[open_period] = {
+                "alert_rule": self.get_alert_rule(open_period),
+                "activities": self.get_incident_activities(open_period),
+            }
+
+        if "activities" in self.expand:
+            # There could be many activities. An incident could seesaw between error/warning for a long period.
+            # e.g - every 1 minute for 10 months
+            for open_period in item_list:
+                results[open_period]["activities"] = self.get_incident_activities(open_period)
+
+        return results
+
+    def get_incident_status(self, priority: PriorityLevel) -> IncidentStatus:
+        return self.priority_to_incident_status[priority]
+
+    def get_incident_activities(
+        self, open_period: GroupOpenPeriod
+    ) -> list[IncidentActivitySerializerResponse]:
+        pass
+
+    def get_alert_rule(self, open_period: GroupOpenPeriod) -> AlertRuleSerializerResponse:
+        pass
```

Comment:
I think this is gonna be kinda gnarly but something like group open period -> group -> actiongroupstatus -> action -> go up the chain until you get the detector data that maps to the alert rule data. You can reference [this](https://www.notion.so/sentry/Alerts-Create-Issues-Metric-Alert-Migration-1198b10e4b5d80e29cdffc191de502c7?pvs=4#11a8b10e4b5d8042b782fb00e793db16) though it might be a bit outdated by now